api/configuration/v1alpha1/kongpluginbinding_types.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2024 Kong, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO(pmalek): expression field is not in OpenAPI spec but error message references it:
+//   when protocols has 'http', at least one of 'hosts', 'methods', 'paths', 'headers' or 'expression' must be set
+
+// KongPluginBinding is the schema for Plugin Bindings API which defines a Kong Plugin Binding.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Plugin-kind",type=string,JSONPath=`.spec.pluginReference.kind`,description="Kind of the plugin"
+// +kubebuilder:printcolumn:name="Plugin-name",type=string,JSONPath=`.spec.pluginReference.name`,description="Name of the plugin"
+// +kubebuilder:printcolumn:name="Kong-Service",type=string,JSONPath=`.spec.kong.serviceReference.name`,description="Service name"
+// +kubebuilder:printcolumn:name="Kong-Route",type=string,JSONPath=`.spec.kong.routeReference.name`,description="Route name"
+// +kubebuilder:printcolumn:name="Kong-Consumer",type=string,JSONPath=`.spec.kong.consumerReference.name`,description="Consumer name"
+// +kubebuilder:printcolumn:name="Kong-ConsumerGroup",type=string,JSONPath=`.spec.kong.consumerGroupReference.name`,description="ConsumerGroup name"
```

Comment:
I'm wondering if we're not locking ourselves in by this. When we add the generic entity ref then print columns will expand even further that that point.

OTOH I see the value of this information being readily available 🤔

Reply:
I agree with you, these fields are certainly useful for the user, but It's probably too much. We can start with just the plugin reference and in case add others later on. WDYT?

Reply:
Sounds good.